pandas/core/frame.py

```diff
@@ -5873,35 +5873,84 @@ def _get_agg_axis(self, axis_num):
 
     def mode(self, axis=0, numeric_only=False):
         """
-        Gets the mode(s) of each element along the axis selected. Adds a row
-        for each mode per label, fills in gaps with nan.
-
-        Note that there could be multiple values returned for the selected
-        axis (when more than one item share the maximum frequency), which is
-        the reason why a dataframe is returned. If you want to impute missing
-        values with the mode in a dataframe ``df``, you can just do this:
-        ``df.fillna(df.mode().iloc[0])``
+        Get the mode(s) of each element along the axis selected. 
+        
+        Adds a row for each mode per label, filling gaps with NaN.
 
         Parameters
         ----------
         axis : {0 or 'index', 1 or 'columns'}, default 0
-            * 0 or 'index' : get mode of each column
-            * 1 or 'columns' : get mode of each row
+            The axis to iterate over while searching for the mode.
+            To find the mode for each column, iterate over rows (``axis=0``, 
+            default behaviour).
+            To find the mode for each row, iterate over columns (``axis=1``).
         numeric_only : boolean, default False
-            if True, only apply to numeric columns
+            If True, only apply to numeric dimensions.
 
         Returns
         -------
         modes : DataFrame (sorted)
+            A DataFrame containing the modes. 
+            If ``axis=0``, there will be one column per column in the original 
+            DataFrame, with as many rows as there are modes.
+            If ``axis=1``, there will be one row per row in the original 
+            DataFrame, with as many columns as there are modes. 
+
+        Notes
+        -----
+        There may be multiple values returned for the selected
+        axis (when more than one item share the maximum frequency), which is
```

Comment:
no parens, just use a comma. capitalize DataFrame.

you can leave off the impute sentence